src/main/kotlin/racing/RacingInputReceiver.kt

```diff
@@ -1,15 +1,32 @@
 package racing
 
-class RacingInputReceiver {
+object RacingInputReceiver {
+
+    private const val MAX_NAME_LENGTH = 5
+    private const val MIN_MOVE_COUNT = 0
 
     @Throws(IllegalArgumentException::class)
     fun receive(): UserInput {
-        println("자동차 대수는 몇 대인가요?")
-        val carNumber = requireNotNull(readLine()?.toInt())
+        val carNames = receiveCarNames()
+        val moveCount = receiveMoveCount()
+        return UserInput(carNames, moveCount)
+    }
 
-        println("시도할 횟수는 몇 회인가요?")
-        val moveCount = requireNotNull(readLine()?.toInt())
+    private fun receiveCarNames(): List<String> {
+        println("경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).")
+        val carNames = requireNotNull(readLine())
+            .replace(" ", "")
+            .split(",")
+
+        require(carNames.none { it.count() > MAX_NAME_LENGTH }) { "이름은 ${MAX_NAME_LENGTH}자 이하로 가능합니다." }
```

Comment:
1. `require` 의 인자로 받은 표현식이 참인 조건과 메시지가 동일하게 표현되는게 더 이해하기 쉬운 것 같습니다.
```suggestion
        require(carNames.all { it.count() <= MAX_NAME_LENGTH }) { "이름은 ${MAX_NAME_LENGTH}자 이하로 가능합니다." }
```
2. `carName` 의 길이는 도메인의 제약사항이라고 생각합니다. 그렇다면 도메인 객체의 생성시점에 검증하는것이 더 나은 선택 아닐까요?